File: cards-service/src/main/java/a/gleb/cardservice/mapper/CardMapper.java
package a.gleb.cardservice.mapper;

import a.gleb.cardservice.db.entity.CardEntity;
import a.gleb.cardservice.model.CardResponse;
import org.springframework.stereotype.Component;

@Component
public class CardMapper {

    /**
     * Mapper to response model.
     * @param entity from database.
     * @return {@link CardResponse} response to API.
     */
    public CardResponse toResponse(CardEntity entity) {
        return CardResponse.builder()
                .cardId(entity.getId())
                .cardTitle(entity.getCardTitle())
                .description(entity.getDescription())
                .userId(entity.getUserId())
                .updatedTime(entity.getLastModifiedDate())
                .build();
    }
}

File: card-service/src/main/kotlin/a/gleb/cardservice/mapper/CardMapper.kt
package a.gleb.cardservice.mapper

import a.gleb.cardservice.db.entity.CardEntity
import a.gleb.cardservice.logger
import a.gleb.cardservice.model.CardRequest
import a.gleb.cardservice.model.CardResponse
import org.springframework.stereotype.Component
import java.time.LocalDateTime
import java.util.*

@Component
class CardMapper {

    fun toResponse(entity: CardEntity): CardResponse {
        logger.info { "Map entity to response: [entity.id=$entity.id]" }
        return with(entity) {
            CardResponse(id, title, description, userId, lastUpdate)
        }
    }

    fun toCardEntity(request: CardRequest, userId: UUID): CardEntity {
        logger.info { "Map request to entity: [request.title=${request.title}, sub=$userId]" }
        return CardEntity(null, request.title, request.description, userId, LocalDateTime.now())
    }
}